src/config.rs

```diff
@@ -94,107 +178,188 @@ pub struct Style {
+    val.parse().map_err(|_| ())
+}
+
+fn parse_attr(val: String) -> u16 {
+    for &(name, attr) in &ATTRS {
+        if name == val {
+            return attr;
+        }
+    }
+
+    return 0;
+}
+
+impl<'de> Deserialize<'de> for Style {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: Deserializer<'de>
+    {
+        #[derive(Deserialize)]
+        #[serde(field_identifier, rename_all = "lowercase")]
+        enum Field { Fg, Bg, Attrs }
+
+        use std::fmt;
+
+        struct StyleVisitor;
+        impl<'de> Visitor<'de> for StyleVisitor {
+            type Value = Style;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                let colors = COLORS.iter().map(|&(name, _)| name).collect::<Vec<&str>>().join(", ");
+                let attrs = ATTRS.iter().map(|&(name, _)| name).collect::<Vec<&str>>().join(", ");
+
+                let expected_format = format!("Expected format:\nfg: 0-255 | colorname\nbg: 0-255 | colorname\nattrs: [{}]\n\nColor Names: {}", attrs, colors);
+                let msg = format!("color style\n{}", expected_format);
+
+                formatter.write_str(msg.as_str())
+            }
+
+            fn visit_map<M>(self, mut map: M) -> Result<Self::Value, M::Error>
+                where M: MapAccess<'de>
+            {
+                let mut fg: Option<u16> = None;
+                let mut bg: Option<u16> = None;
+                let mut attr: u16 = 0;
+
+                while let Some(key) = map.next_key()? {
+                    match key {
+                        Field::Fg => {
+                             let color = parse_color(map.next_value()?) 
+                                 .map_err(|_| de::Error::invalid_value(de::Unexpected::UnitVariant, &self))?;
+
+                             fg = Some(color);
+                        },
+
+                        Field::Bg => {
+                            let color = parse_color(map.next_value()?)
+                                .map_err(|_| de::Error::invalid_value(de::Unexpected::UnitVariant, &self))?;
+
+                            bg = Some(color);
+                        },
+
+                        Field::Attrs => {
+                            let attrs: Vec<String> = map.next_value()?;
+                            attr = attrs.into_iter()
+                                .map(parse_attr)
+                                .fold(0, |style, a| style | a);
+                        }
+                    }
+                }
+
+                let fg = fg.ok_or_else(|| de::Error::missing_field("fg"))?;
+                let bg = bg.ok_or_else(|| de::Error::missing_field("bg"))?;
+
+                Ok(Style { fg: fg | attr, bg })
+            }
+        }
+
+        deserializer.deserialize_map(StyleVisitor)
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(default)]
+pub struct Theme {
+    /// Colors used to highlight nicks
+    pub nick_colors: [u8; 14],
```

Comment:
I think we shouldn't fix this to 14 colors, users may want to have more or less colors here so maybe we should use a `Vec<u8>`.

One problem with that is we won't be able to use `default_theme()` function for initialization anymore because we can't allocate a vector in a const function.